trace_http_client.go

```diff
@@ -20,18 +20,36 @@ type HTTPClient interface {
 type TraceHTTPClient struct {
 	client HTTPClient
 	tracer opentracing.Tracer
+	withUserErr bool
```

Comment:
Done!

And agree with `includeClientErrors`.  It's a tricky one to name since it's flagging whether to set errors as errors, just in a different context.